Update install instructions for AM 1.7 (including headless/indexless installations)

jrwdunham opened the pull request:
Most importantly, this instructs users to set the `_SEARCH_ENABLED`-suffixed env vars to `'false'` if they want to run Archivematica "headless", i.e., with Elasticsearch not installed. This commit also includes some formatting and spelling changes.

This also removes the section entitled `Upgrade from Archivematica 1.8.0 to 1.8` because it did not seem to make sense. This section may need to be restored and re-written.

Note that the links to the install/README.md files (e.g., https://github.com/artefactual/archivematica/blob/stable/1.7.x/src/MCPServer/install/README.md) should reference the `_SEARCH_ENABLED` env vars once the headless PR is merged into AM's stable/1.7.x.

I created a Xenial VM and followed these instructions and they work, except that the https://packages.archivematica.org/1.7.x/ type URLs are pointing to 1.6.1 packages, which resulted in the `_SEARCH_ENABLED` env vars having no effect. These instructions need to be re-tested once AM PR 901 is merged and new packages are created for AM 1.7.

jhsimpson committed:
Updating install docs to describe a headless option.

First draft, incomplete

refs #33

kellyannewithane committed:
Add instructions for Elasticsearch as optional

kellyannewithane committed:
Update to use jenkins-ci instead of packages.archivematica

kellyannewithane committed:
Install instructions with optional elasticsearch

sromkey committed:
Fixed link to gpg encryption docs in Storage Service

scollazo committed:
Upgrade install instructions for 1.8.x release

scollazo committed:
Update gearmand restart command

scollazo committed:
Enable fits service, so it starts at boot

sevein committed:
Update release and version

scollazo committed:
Add notes regarding ssl

scollazo committed:
Minor fixes noted by @sevein

jrwdunham committed:
Modify 1.7 install instructions for headless

Most importantly, this instructs users to set the `_SEARCH_ENABLED`-suffixed
env vars to `'false'` if they want to run Archivematica "headless", i.e., with
Elasticsearch not installed. This commit also includes some formatting changes.

Also removes the section entitled `Upgrade from Archivematica 1.8.0 to 1.8`
because it did not seem to make sense. This section may need to be restored and
re-written.

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -75,35 +73,42 @@ however, processing large collections will require better hardware. See
 
-**Notes**:
+.. note:: *Indexing*. Installing Elasticsearch to provide a search index is now
+   optional as of version |release|.  Installing Archivematica in a headless state,
```
Why this is called "headless"? We still expect users to use the web interface, afaik.

jrwdunham replied:
You're right. I believe that this "remove Elasticsearch" (a.k.a. "indexless") work is heading in the direction of being able to run Archivematica in a completely "headless" way, such that you could do everything via the API and never even run the dashboard GUI. But clearly we are not there yet, so maybe we should just call this "indexless" until such time as we really have headless functionality.

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -240,14 +255,26 @@ get the most recent version of pip.
 
+If you are planning running Archivematica in headless mode (i.e., without
+Elasticsearch), then set the relevant environment variables to ``false``. (See
+the `Dashboard install README`_, the `MCPClient install README`_, and the
```
Can this be rephrased? The READMEs are pertinent always, not only for installs without ES

jrwdunham replied:
Good point. I'll rephrase as you say.

scollazo reviewed (changes requested):
This might sound dumb, but why installs without ES are being called "headless"?

jrwdunham committed:
Change "headless" to "indexless" in install docs

Also updated the installation.rst document so that the links to the
install/README files are made not just in the context of performing an
indexless deploy.

jrwdunham committed:
Fix indexless env var instructions given systemd

jrwdunham commented:
@scollazo I've addressed your change requests and updated the indexless env var instructions to work correctly with systemd.

jrwdunham committed:
Remove `systemctl set-environment` instructions

jrwdunham commented:
@sevein yeah, I agree. I removed the `set-environment` stuff.

sevein commented on admin-manual/installation/installation.rst:
```diff
@@ -167,27 +172,60 @@ Archivematica packages are hosted at packages.archivematica.org. This has been
 
 Using 14.04 (Trusty):
 
+Run these three commands right now (**Delete this section when the final release
```
`s/Delete/delete`?

sevein commented on admin-manual/installation/installation.rst:
```diff
@@ -167,27 +172,60 @@ Archivematica packages are hosted at packages.archivematica.org. This has been
 
 Using 16.04 (Xenial):
 
+Run these three commands right now (**Delete this section when the final release
```
`s/Delete/delete`?

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -167,27 +172,60 @@ Archivematica packages are hosted at packages.archivematica.org. This has been
+
+.. code:: bash
+
+   sudo wget -O - https://packages.archivematica.org/1.7.x/key.asc | sudo apt-key add -
```
I think we should add the urls for keys and repos to the conf.py, so we have one central place to do all the changes needed for the release.

jrwdunham replied:
@scollazo I think I tried doing that originally and I had some trouble getting sphinx to do that interpolation in code blocks... I'll try again and report back

scollazo replied:
Ok,if it's too hard forget about it. May be ading the full block might make it easier?

jrwdunham replied:
ok, looks like it works if you use `.. parsed-literal::` instead of `.. code:: bash`. As long as we're not relying on code highlighting for bash commands in our docs, I think that'll be ok. See https://stackoverflow.com/questions/25557602/sphinx-substitutions-next-to-angle-brackets/25557867#25557867.

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -349,15 +411,18 @@ Some repositories need to be installed in order to fullfill the installation pro
 
-Common services like elasticsearch, mariadb and gearmand should be installed
-and enabled before the archivematica install. It can be done with:
+Common services like Elasticsearch, mariadb and gearmand should be installed
```
/mariadb/MariaDB/ and gearmand/Gearmand

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -371,13 +436,13 @@ and enabled before the archivematica install. It can be done with:
    sudo -u root yum install -y python-pip archivematica-storage-service
 
-* After the package is installed, we need to populate the sqlite database, and
+* After the package is installed, populate the sqlite database, and
```
sqlite/SQLite

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -478,19 +533,40 @@ and enabled before the archivematica install. It can be done with:
+
+If you are planning on running Archivematica in indexless mode (i.e., without
+Elasticsearch), then modify the relevant systemd EnvironmentFile files by
+adding lines that set the relevant environment variables to ``false``:
```
A note like "Remember to stop elasticsearch if you don't want to use it, to free some resources" would be handy

jrwdunham replied:
If the user is deploying an indexless Archivematica, then presumably they will have listened to the notes above this and not installed or started it in the first place. Therefore, I don't think that a note like you describe makes sense here. If you disagree, please let me know.

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -753,7 +846,9 @@ better to update the dashboard before updating the mcp components.
 7. Disable Unused Services
 
-Archivematica 1.6.0 uses nginx as http server, and gunicorn as wsgi server. This means that some services used in Archivematica 1.5.0 should be stopped and disabled before performing the upgrade.
+Archivematica |release| uses nginx as http server, and gunicorn as wsgi server.
```
Nginx and Gunicorn

scollazo replied:
Wsgi too?

jrwdunham replied:
I think it's WSGI

scollazo reviewed (changes requested):
Some minor fixes, sorry for the nitpicking ;)

scollazo commented on admin-manual/installation/installation.rst:
```diff
@@ -386,27 +451,31 @@ and enabled before the archivematica install. It can be done with:
    ";
 
-* And now, we enable and start the archivematica-storage-service and it’s nginx frontend
+* Now enable and start the archivematica-storage-service, rngd (needed for encrypted spaces) and the nginx frontend:
```
Rngd?

jrwdunham replied:
Googling rngd, I don't see any capitalized variants, so I'm gonna leave it lowercase.

jrwdunham committed:
Use sphinx substitutions for URLs

Also fixed some misspellings noticed in CR.

jrwdunham commented:
Closing in favour of https://github.com/artefactual/archivematica-docs/pull/73
